gui/wxpython/startup/guiutils.py

```diff
@@ -33,30 +31,62 @@ def SetSessionMapset(database, location, mapset):
     RunCommand("g.gisenv", set="MAPSET=%s" % mapset)
 
 
-
 class NewMapsetDialog(TextEntryDialog):
     def __init__(self, parent=None, default=None,
-                 validation_failed_handler=None, help_hanlder=None):
-        if help_hanlder:
-            style = wx.OK | wx.CANCEL | wx.HELP
-        else:
-            style = wx.OK | wx.CANCEL
-        if validation_failed_handler:
-            validator=GenericValidator(
-                gs.legal_name, validation_failed_handler)
-        else:
-            validator = None
+                 database=None, location=None):
+        self.database = database
+        self.location = location
+
+        # list of tuples consisting of conditions and callbacks
+        checks = [(gs.legal_name, self._nameValidationFailed),
+                  (self._checkMapsetNotExists, self._mapsetAlreadyExists),
+                  (self._checkOGR, self._reservedMapsetName)]
+        validator = GenericMultiValidator(checks)
+
         TextEntryDialog.__init__(
             self, parent=parent,
             message=_("Name for the new mapset:"),
             caption=_("Create new mapset"),
             defaultValue=default,
             validator=validator,
-            style=style
         )
-        if help_hanlder:
-            help_button = self.FindWindowById(wx.ID_HELP)
-            help_button.Bind(wx.EVT_BUTTON, help_hanlder)
+
+    def _nameValidationFailed(self, ctrl):
+        message = _(
+            "Name <%(name)s> is not a valid name for location or mapset. "
+            "Please use only ASCII characters excluding %(chars)s "
+            "and space.") % {
+            'name': ctrl.GetValue(),
+            'chars': '/"\'@,=*~'}
+        GError(parent=self, message=message, caption=_("Invalid name"))
+
+    def _checkOGR(self, text):
+        """Check user's input for reserved mapset name."""
+        if text.lower() == 'ogr':
+            return False
+        return True
+
+    def _reservedMapsetName(self, ctrl):
+        message = _(
+            "Name <%s> is reserved for direct "
+            "read access to OGR layers. Please use "
+            "another name for your mapset.") % {
+             ctrl.GetValue()}
+        GError(parent=self, message=message,
+               caption=_("Reserved mapset name"))
+
+    def _checkMapsetNotExists(self, text):
+        """Check whether user's input mapset exists or not."""
+        if mapset_exists(self.database, self.location, text):
+            return False
+        return True
+
+    def _mapsetAlreadyExists(self, ctrl):
+        message = _(
+            "Mapset <%s> already exists. Please consider to use "
+            "another name for your location.") % {
+             ctrl.GetValue()}
```

Comment:
This is obviously not right, please fix again. Use format function.
![Existing mapset path_020](https://user-images.githubusercontent.com/7494312/86130899-e197fa00-bab2-11ea-80c1-afd4cbdf1df2.png)